test/parallel/test-stream-unpipe-event.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const {Writable, Readable} = require('stream');
+class NullWriteable extends Writable {
+  _write(chunk, encoding, callback) {
+    return callback();
+  }
+}
+class QuickEndReadable extends Readable {
+  _read() {
+    this.push(null);
+  }
+}
+class NeverEndReadable extends Readable {
+  _read() {}
+}
+
+{
+  const dest = new NullWriteable();
+  const src = new QuickEndReadable();
+  dest.on('pipe', common.mustCall(() => {}));
+  dest.on('unpipe', common.mustCall(() => {}));
+  src.pipe(dest);
+  setImmediate(() => {
+    assert.strictEqual(src._readableState.pipesCount, 0);
+  });
+}
+
+{
+  const dest = new NullWriteable();
+  const src = new NeverEndReadable();
+  dest.on('pipe', common.mustCall(() => {}));
```

Comment:
Since this PR was opened, there were some changes to `common`. We have `common.noop` now for this type of callback. `common.mustCall()` can be called with no function, and it will default to `common.noop`.

TL;DR - `common.mustCall(() => {})` throughout this PR can be replaced with `common.mustCall()`.